src/Data/Text/Rope.hs

```diff
@@ -376,25 +360,7 @@ subOnLines tl (Position xl xc) (Position yl yc) = case xl `compare` yl of
 -- ("f\n𐀀я","")
 --
 splitAtPosition :: HasCallStack => Position -> Rope -> (Rope, Rope)
-splitAtPosition (Position 0 0) = (mempty,)
-splitAtPosition !len = \case
-  Empty -> (Empty, Empty)
-  Node l c r _
-    | len <= ll -> case splitAtPosition len l of
-      (before, after)
-        | null after -> case splitAtPosition len' (c <| r) of
-          (r', r'') -> (l <> r', r'')
-        | otherwise -> (before, node after c r)
-    | len <= llc -> case TL.splitAtPosition len' c of
-      (before, after)
-        | TL.null after -> case splitAtPosition len'' r of
-          (r', r'') -> ((l |> c) <> r', r'')
-        | otherwise -> (l |> before, after <| r)
-    | otherwise -> case splitAtPosition len'' r of
-      (before, after) -> (node l c before, after)
-    where
-      ll = lengthAsPosition l
-      lc = TL.lengthAsPosition c
-      llc = ll <> lc
-      len' = subOnRope l len ll
-      len'' = subOnLines c len' lc
+splitAtPosition (Position l c) rp = (beforeLine <> beforeColumn, afterColumn)
```

Comment:
It's surprising that there is no performance degradation: I'd expect that `splitAtPosition` was implemented in such way to avoid extra concatenations.

Reply:
This was also a bit surprising to me. I've tried a few variations that avoided this but they turned out to be both slower (at least according to the current benchmarks) and more complicated code-wise.

Might be worth revisiting though.